### PR TITLE
#2915 | queryWithCursor - make recursive calls till limit reached, when there are filter are applied on result set

### PR DESCRIPTION
**Pull request by sravan1000**
## Description

- [X] Query with Cursor - Recursive calls till limit reached
## TODO
- [ ] ~handle this case - when applied filter has only two items, and when limit is set to 5~

## Related Issues

Issue: #2915

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

File: src/query/queryWithCursor.ts [hunk 1]
```diff
@@ -39,15 +39,36 @@ export async function queryWithCursor<T extends AnyObject>(
...
-    LastEvaluatedKey,
-    Items,
-    ScannedCount,
-  }: QueryOutput = await dbClient.query(params).promise();
+  let itemsData = [];
+  let LastEvaluatedKeyOutput: Key = undefined;
+  let canIterate = true;
+  let scannedCountCopy = 0;
+
+  do {
```
Comment (snehithv) [inline src/query/queryWithCursor.ts:47]: you need to do this only if `filter.where` is not empty

Comment (sravan1000) [reply src/query/queryWithCursor.ts:47]: or i can add condition in while loop

`while (LastEvaluatedKeyOutput && canIterate && params?.Limit ) ;`

Comment (sravan1000) [reply src/query/queryWithCursor.ts:47]: Changed, Now it will iterate only when there is `limit` in `params`

File: src/query/queryWithCursor.ts [hunk 2]
```diff
@@ -39,15 +39,36 @@ export async function queryWithCursor<T extends AnyObject>(
...
+    const {
+      LastEvaluatedKey,
+      Items,
+      ScannedCount,
+    }: QueryOutput = await dbClient.query(params).promise();
+
+    LastEvaluatedKeyOutput = LastEvaluatedKey;
+
+    scannedCountCopy = ScannedCount;
```
Comment (snehithv) [inline src/query/queryWithCursor.ts:56]: you need to increment this every time this loops run

Comment (sravan1000) [reply src/query/queryWithCursor.ts]: Changed by lakhmaji

**Review by snehithv — APPROVED** [review]

**Pull request merged**